src/components/Select/Select.tsx

```diff
@@ -58,19 +70,32 @@ const Select = React.forwardRef(
       [`${className}__wrapper`]: className,
     });
 
+    const selectClassnames = classnames(`${px}-input__input`, {
+      className,
+      [`${px}-input__select--open`]: isOpen && showCustomIcon,
+      [`${px}-input__select--closed`]: !isOpen && showCustomIcon,
+      [`${px}-input__select--tertiary`]: variant === SelectVariants.tertiary,
+    });
+
+    const handleClick = (e: React.MouseEvent<HTMLSelectElement>) => {
+      handleIsOpen();
+      onClick?.(e);
+    };
+
     return (
       <div className={wrapperClassnames}>
         <label htmlFor={id} className={classnames(`${px}-input__label`, { [`${px}-input__label--hidden`]: hideLabel })}>
           {labelText}
         </label>
         <select
-          className={classnames(`${px}-input__input`, { className })}
+          className={selectClassnames}
           data-testid={id}
           defaultValue={defaultValue}
           disabled={inputProps.disabled}
           id={id}
           onChange={onChange}
-          onClick={onClick}
+          onBlur={closeDropdown}
```

Comment:
isn't opening and closing handled already by the standard select HTML code.  Why do we need to have our own `isOpen` state locally here in the Select component?

Reply:
is it possible to apply our custom icon styles using a CSS attribute selector on the Select.  For instance `:focus` pseudo selector.  As much as possible we don't want any internal state in our Seldon components

Reply:
@scottdickerson we need it for icon's rotation, html's select has not any `open` state or attribute